src/utilities.ts

```diff
@@ -52,28 +53,35 @@ interface ExecuteWithMetricsParams<T extends string> {
   count?: Counter<T>
 }
 
-export async function executeWithMetrics<T extends string>(params: ExecuteWithMetricsParams<T>) {
-  const { runnable, totalCount, count, errorsCount, durationsTotals, durations, labelValues } = params
+export function withMetrics<T extends string>(params: MetricsParams<T>) {
+  const { totalCount, count, errorsCount, durationsTotals, durations, labelValues } = params
 
-  // Total count, and count by labels
-  totalCount.inc()
-  if (count && labelValues) {
-    count.inc(labelValues)
-  }
-  const endTimerTotal = durationsTotals.startTimer()
-  const endTimer = durations?.startTimer()
-  try {
-    // Run
-    await runnable()
-  } catch (error) {
-    // Count errors and rethrow
-    errorsCount.inc()
-    throw error
-  } finally {
-    // End timers
-    endTimerTotal()
-    if (endTimer && labelValues) {
-      endTimer(labelValues)
+  return <T extends (...args: any[]) => any>(runnable: T) =>
+    async (...params: Parameters<T>): Promise<ReturnType<T>> => {
+      // Total count, and count by labels
+      totalCount.inc()
+      if (count && labelValues) {
+        count.inc(labelValues)
+      }
+      const endTimerTotal = durationsTotals.startTimer()
+      const endTimer = durations?.startTimer()
+      try {
+        // Run
+        return await runnable(...params)
+      } catch (error) {
+        // Count errors and rethrow
+        errorsCount.inc()
+        throw error
+      } finally {
+        // End timers
+        endTimerTotal()
+        if (endTimer && labelValues) {
+          endTimer(labelValues)
+        }
+      }
     }
-  }
 }
+
+export const withOrderBookMetrics = withMetrics(orderBookMetrics)
+export const withBuyAmountEstimationMetrics = withMetrics(buyAmountEstimationMetrics)
+export const withOrderBookFetcherMetrics = withMetrics(orderBookFetcher)
```

Comment:
I think this doesn't belong here.

HOF function, yes, that is a utility.
But the actual function combined with the metrics should either be exported from where the metrics is defined or where it's used.